bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/handler/SiemensHvacBridgeBaseThingHandler.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.siemenshvac.internal.handler;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.siemenshvac.internal.discovery.SiemensHvacDeviceDiscoveryService;
+import org.openhab.binding.siemenshvac.internal.metadata.SiemensHvacMetadataRegistry;
+import org.openhab.binding.siemenshvac.internal.network.SiemensHvacConnector;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.net.NetworkAddressService;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.types.Command;
+
+/**
+ * The {@link SiemensHvacBridgeBaseThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Laurent Arnal - Initial contribution and API
+ */
+@NonNullByDefault
+public abstract class SiemensHvacBridgeBaseThingHandler extends BaseBridgeHandler {
+
+    private @Nullable SiemensHvacDeviceDiscoveryService discoveryService;
+    private final @Nullable HttpClientFactory httpClientFactory;
+    private final SiemensHvacMetadataRegistry metaDataRegistry;
+    private @Nullable SiemensHvacBridgeConfig config;
+
+    public SiemensHvacBridgeBaseThingHandler(Bridge bridge, @Nullable NetworkAddressService networkAddressService,
+            @Nullable HttpClientFactory httpClientFactory, SiemensHvacMetadataRegistry metaDataRegistry) {
+        super(bridge);
+        SiemensHvacConnector lcConnector = null;
+        this.httpClientFactory = httpClientFactory;
+        this.metaDataRegistry = metaDataRegistry;
+
+        lcConnector = this.metaDataRegistry.getSiemensHvacConnector();
+        if (lcConnector != null) {
+            lcConnector.setSiemensHvacBridgeBaseThingHandler(this);
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SiemensHvacBridgeConfig.class);
+
+        metaDataRegistry.readMeta();
+    }
+
+    public @Nullable SiemensHvacBridgeConfig getBridgeConfiguration() {
+        return config;
+    }
+
+    @Override
+    public void updateStatus(ThingStatus status) {
+        super.updateStatus(status);
+    }
```

Comment:
I removed it on next commit

Reply:
Still there (line 135-138).

Reply:
this one is need because of call from SiemensHvacHandlerImpl. line 157 / 174.

Reply:
Ah, you're exposing it as `public` so that child handlers can directly change the status of the Bridge. If one Thing handler has an errorRate > 50, and another one doesn't, won't that cause the Bridge to continuously flip status between ONLINE and OFFLINE?

Reply:
Yes.

I'm not sure that the flip status you describe could happen.

The errorRate is mainly there today two handle network communication errors between the binding and the ozw672 gateway.

If the error is occuring between the ozw672 and the slave device like the rvsxxx, i don't think it will it this code because the reading of value between ozw672 and rvs is done asynchronously, and i'm not sure it will toggle an error at this level.

I will need to test it to verify, disconnecting the RVS from the OZW to verify what happens in the case two be sure.

Also I have only one slave device on my home setup, so not very simple two verify such multiple device case !